src/com/facebook/buck/jvm/java/JavaSourceJar.java

```diff
@@ -81,12 +99,21 @@ public JavaSourceJar(
     steps.add(new MakeCleanDirectoryStep(getProjectFilesystem(), temp));
 
     Set<Path> seenPackages = Sets.newHashSet();
+    Map<ProjectFilesystem, JavaPackageFinder> finders = new HashMap<>();
+
+    for (SourcePath sourcePath : sources) {
+      Path source = getResolver().getAbsolutePath(sourcePath);
+      if (!"java".equals(MorePaths.getFileExtension(source))) {
```

Comment:
should make the list of extensions configurable as other jvm based libraries can have scala/groovy/kotlin files etc and can easily override this